Casbin/Util/BuiltInFunctions.cs

```diff
@@ -128,6 +128,30 @@ public static bool KeyMatch4(string key1, string key2)
 
             return true;
         }
+        /// <summary>
+        /// KeyMatch determines whether key1 matches the pattern of key2 and ignores the parameters in key2.
+        /// For example, "/foo/bar?status=1&type=2" matches "/foo/bar"
+        /// </summary>
+        /// <param name="key1"> The first argument. </param>
+        /// <param name="key2"> The second argument. </param>
+        /// <returns></returns>
+        public static bool KeyMatch5(string key1, string key2)
+        {
+            int index = key1.IndexOf('?');
+#if NETFRAMEWORK
+            if (index == -1)
+            {
+                return key1.Equals(key2);
+            }
+            return key1.Substring(0, index).Equals(key2);
+#else
+            if (index == -1)
+            {
+                return key1.AsSpan().Equals(key2.AsSpan(), StringComparison.Ordinal);
+            }
+            return key1.AsSpan().Slice(0, index).Equals(key2.AsSpan(), StringComparison.Ordinal);
+#endif
+        }
```

Comment:
Just a suggestion, `key1.AsSpan()` repeated twice is not so readable and for the constants, I recommend using `is` instead of `==` operator.

```suggestion
        /// <summary>
        /// KeyMatch determines whether key1 matches the pattern of key2 and ignores the parameters in key2.
        /// For example, "/foo/bar?status=1&amp;type=2" matches "/foo/bar"
        /// </summary>
        /// <param name="key1"> The first argument. </param>
        /// <param name="key2"> The second argument. </param>
        /// <returns></returns>
        public static bool KeyMatch5(string key1, string key2)
        {
            var key1Span = key1.AsSpan();
            var key2Span = key2.AsSpan();
            int index = key1Span.IndexOf('?');
            return index is -1
                ? key1Span.Equals(key2Span, StringComparison.Ordinal)
                : key1Span.Slice(0, index).Equals(key2Span, StringComparison.Ordinal);
        }
```

Reply:
Thank you for the suggestion! I'll force-pushed these changes.